Sources/ZcashLightClientKit/Synchronizer/SDKSynchronizer.swift

```diff
@@ -652,6 +652,173 @@ public class SDKSynchronizer: Synchronizer {
+            let url: String
+        }
+        
+        struct CheckResult {
+            let id: String
+            let info: LightWalletdInfo?
+            let getInfoTime: TimeInterval
+            let latestBlockHeight: BlockHeight?
+            let latestBlockHeightTime: TimeInterval
+            let mean: TimeInterval
+            let service: Service
+            var blockTime: TimeInterval
+        }
+        
+        // Initialize services for the endpoints
+        let services = endpoints.map {
+            Service(
+                originalEndpoint: $0,
+                service: LightWalletGRPCService(
+                    host: $0.host,
+                    port: $0.port,
+                    secure: $0.secure,
+                    singleCallTimeout: 5000,
+                    streamingCallTimeout: Int64(fetchThreshold) * 1000
+                ),
+                url: "\($0.host):\($0.port)"
+            )
+        }
+
+        // Parallel part
+        var checkResults: [String: CheckResult] = [:]
+        
+        await withTaskGroup(of: CheckResult.self) { group in
+            for service in services {
+                group.addTask {
+                    let startTime = Date().timeIntervalSince1970
+                    let info = try? await service.service.getInfo()
+                    let markTime = Date().timeIntervalSince1970
+                    let latestBlockHeight = try? await service.service.latestBlockHeight()
+                    let endTime = Date().timeIntervalSince1970
+
+                    let getInfoTime = markTime - startTime
+                    let latestBlockHeightTime = endTime - markTime
+                    let mean = (getInfoTime + latestBlockHeightTime) / 2
+
+                    return CheckResult(
+                        id: service.url,
+                        info: info,
+                        getInfoTime: getInfoTime,
+                        latestBlockHeight: latestBlockHeight,
+                        latestBlockHeightTime: latestBlockHeightTime,
+                        mean: mean,
+                        service: service,
+                        blockTime: 0
+                    )
+                }
+            }
+            
+            var tmpResults: [String: CheckResult] = [:]
+            
+            for await result in group {
+                // rule out results where calls failed
+                guard let info = result.info, let _ = result.latestBlockHeight else {
+                    continue
+                }
+                
+                // rule out if mismatch of networks
+                guard (info.chainName == "main" && network == .mainnet) 
+                    || (info.chainName == "test" && network == .testnet) else {
+                    continue
+                }
+                
+                // rule out mismatch of consensus branch IDs
+                if let localBranchID = await blockProcessor.consensusBranchIdFor(Int32(info.blockHeight)) {
```

Comment:
```suggestion
                guard let localBranchID = await blockProcessor.consensusBranchIdFor(Int32(info.blockHeight)) else {
                    continue
                }
```
(Just a style issue, non-blocking.)

Reply:
Updated